plugin-build/src/main/kotlin/io/sentry/android/gradle/instrumentation/androidx/compose/ComposeNavigation.kt

```diff
@@ -0,0 +1,76 @@
+package io.sentry.android.gradle.instrumentation.androidx.compose
+
+import com.android.build.api.instrumentation.ClassContext
+import io.sentry.android.gradle.instrumentation.ClassInstrumentable
+import io.sentry.android.gradle.instrumentation.SpanAddingClassVisitorFactory
+import io.sentry.android.gradle.instrumentation.wrap.Replacement
+import org.objectweb.asm.ClassVisitor
+import org.objectweb.asm.MethodVisitor
+import org.objectweb.asm.Type
+import org.objectweb.asm.commons.AdviceAdapter
+import org.objectweb.asm.commons.Method
+
+open class ComposeNavigation : ClassInstrumentable {
+
+    companion object {
+        private const val NAV_HOST_CONTROLLER_CLASSNAME = "androidx.navigation.compose.NavHostControllerKt"
+        private const val REMEMBER_NAV_CONTROLLER_NAME = "rememberNavController"
+
+        /* ktlint-disable max-line-length */
+        val replacement = Replacement(
+            "Lio/sentry/compose/SentryNavigationIntegrationKt;",
+            "withSentryObservableEffect",
+            "(Landroidx/navigation/NavHostController;Landroidx/compose/runtime/Composer;I)Landroidx/navigation/NavHostController;"
+        )
+        /* ktlint-enable max-line-length */
+    }
+
+    override fun getVisitor(
+        instrumentableContext: ClassContext,
+        apiVersion: Int,
+        originalVisitor: ClassVisitor,
+        parameters: SpanAddingClassVisitorFactory.SpanAddingParameters
+    ): ClassVisitor {
+        return object : ClassVisitor(apiVersion, originalVisitor) {
```

Comment:
`m`: can we return a `CommonClassVisitor` here instead? It has built-in logging of the instrumentation, so we'd just need to provide a MethodInstrumentable for it (needs a bit of refactoring here). You can check how I've done it, for example, for `OkHttp`  -> https://github.com/getsentry/sentry-android-gradle-plugin/blob/main/plugin-build/src/main/kotlin/io/sentry/android/gradle/instrumentation/okhttp/OkHttp.kt

Also, if possible, can we adhere to the package structure of others, and put the actual MethodVisitor implementation under the `androidx.compose.visitor` package? Thanks 👍